locals.tf

```diff
@@ -79,4 +79,8 @@ locals {
       }
     }
   }
+}
+
+locals {
+  network_resource_group_id = regex("(.*?/resourceGroups/[^/]+)", var.network.node_subnet_id)[0]
```

Comment:
might we have a scenario where `var.network.node_subnet_id` is going to be empty? should we add a check for this?